src/test/java/org/springframework/data/mapping/model/BasicPersistentEntityUnitTests.java

```diff
@@ -276,6 +283,64 @@ public void getRequiredAnnotationThrowsException() {
 		assertThatThrownBy(() -> entity.getRequiredAnnotation(Document.class)).isInstanceOf(IllegalStateException.class);
 	}
 
+	@Test // DATACMNS-1210
+	public void findAnnotationShouldBeThreadSafe() throws InterruptedException {
+
+		assumeTrue("Requires Java 9",
```

Comment:
What's the reason we have to assume Java 9 here?

Reply:
I can remove it - the point was that the test would pass on Java8 regardless of the changes made.

Reply:
Okay, I'll just go ahead and remove it during the merge.